src/Core/Containers/VectorArray.hpp

```diff
@@ -26,68 +39,30 @@ class VectorArray : public AlignedStdVector<V>
     /// Returns the array as an Eigen Matrix Map
     MatrixMap getMap() {
         CORE_ASSERT( !this->empty(), "Cannot map an empty vector " );
-        return MatrixMap( this->data()->data(), V::RowsAtCompileTime, this->size() );
+        return MatrixMap(
+            TypeHelper::getData( this ), TypeHelper::Size, Eigen::Index( this->size() ) );
     }
 
     /// Returns the array as an Eigen Matrix Map (const version)
     ConstMatrixMap getMap() const {
         CORE_ASSERT( !this->empty(), "Cannot map an empty vector " );
-        return ConstMatrixMap( this->data()->data(), V::RowsAtCompileTime, this->size() );
+        return MatrixMap(
+            TypeHelper::getConstData( this ), TypeHelper::Size, Eigen::Index( this->size() ) );
     }
 };
-
-/// This specialization stores an array of scalars which can be used as a dynamic
-/// Eigen column vector.
-template <>
-class VectorArray<float> : public AlignedStdVector<float>
-{
-  public:
-    // Type shortcuts
-    using Matrix         = Eigen::Matrix<float, 1, Eigen::Dynamic>;
-    using MatrixMap      = Eigen::Map<Matrix>;
-    using ConstMatrixMap = Eigen::Map<const Matrix>;
-
-  public:
-    /// Inheriting constructors from std::vector
-    using AlignedStdVector<float>::AlignedStdVector;
-
-    /// Returns the array as an Eigen Matrix Map
-    MatrixMap getMap() {
-        CORE_ASSERT( !this->empty(), "Cannot map an empty vector " );
-        return MatrixMap( this->data(), 1, Eigen::Index( this->size() ) );
-    }
-
-    /// Returns the array as an Eigen Matrix Map (const version)
-    ConstMatrixMap getMap() const {
-        CORE_ASSERT( !this->empty(), "Cannot map an empty vector " );
-        return ConstMatrixMap( this->data(), 1, Eigen::Index( this->size() ) );
-    }
+template <typename V>
+struct VectorArrayTypeHelper<V, true> {
+    using Scalar              = V;
+    static constexpr int Size = 1;
```

Comment:
```suggestion
    static constexpr int NumberOfComponents = 1;
```

Reply:
To prevent openGL binding for non arithmetic types using the number of compoenents, we should also have something like 
```suggestion
    static constexpr int NumberOfComponents = std::is_arithmetic_v<V> ? 1 : 0;
```
optinally, we could try to allow also some enum bindings (e.g using the enums underlying type as value_type instead of the enum type itself ...